pandas/tests/io/parser/test_common.py

```diff
@@ -2135,3 +2135,17 @@ def test_no_header_two_extra_columns(all_parsers):
     parser = all_parsers
     df = parser.read_csv(stream, header=None, names=column_names, index_col=False)
     tm.assert_frame_equal(df, ref)
+
+
+def test_dtype_with_parse_dates(all_parsers):
+    # GH 34066
+    parser = all_parsers
+    data = """
+a,b
+1,2020-05-23 01:00:00"""
+    expected = DataFrame(
+        [["1", "2020-05-23 01:00:00"]], columns=["a", "b"], dtype="string"
+    )
+    expected = expected.astype({"b": np.datetime64})
```

Comment:
this is very strange to astype this way, please use `pd.to_datetime`

Reply:
@dsaxton's suggestion [above](https://github.com/pandas-dev/pandas/pull/34330#discussion_r429962579) to declare types in the constructor would remove the need to do any type-casting.